dds/DCPS/XTypes/DynamicDataImpl.cpp

```diff
@@ -5962,6 +6000,640 @@ bool operator<<(Serializer& ser, const KeyOnly<DDS::DynamicData_ptr>& key)
   return serialize(ser, key.value, Sample::KeyOnly);
 }
 
+} // namespace DCPS
+
+namespace DCPS {
+
+bool check_rc(DDS::ReturnCode_t rc, DDS::MemberId id, DDS::TypeKind tk, const char* fn_name)
+{
+  return check_rc_from_get(rc, id, tk, fn_name, LogLevel::Warning);
+}
+
+void vwrite(ValueWriter& vw, DDS::DynamicData_ptr value);
+
+void vwrite_item(ValueWriter& vw, DDS::DynamicData_ptr value,
+                 DDS::MemberId id, const DDS::DynamicType_var& item_type)
+{
+  using namespace XTypes;
+  const DDS::TypeKind item_tk = item_type->get_kind();
+  DDS::TypeKind tk = item_tk;
+
+  DDS::ReturnCode_t rc = DDS::RETCODE_OK;
+  if (item_tk == TK_ENUM) {
+    rc = enum_bound(item_type, tk);
+    if (rc != DDS::RETCODE_OK) {
+      if (log_level >= LogLevel::Warning) {
+        ACE_ERROR((LM_WARNING, "(%P|%t) WARNING: vwrite_item: enum_bound failed (%C)\n",
+                   retcode_to_string(rc)));
+      }
+      return;
+    }
+  }
+  if (item_tk == TK_BITMASK) {
```

Comment:
This could be an `else if` but the compiler is probably going to figure that out on its own -- I'd prefer having it in there so that readers realize that it can't take both "then" branches.